code_map.py

```diff
@@ -816,16 +780,19 @@ def on_load(self, view):
     # -----------------
 
     def on_close(self, view):
+        '''Strangely this block isn't run when closing CodeMap through
+        command, that is following the command w.run_command("close_file"), so
+        here we're just checking if CodeMap has been closed with mouse'''
 
```

Comment:
this was the problem

Reply:
Yeah, I also noticed that. 

It is in fact a strange gap in the ST API. You can open a document with `open_file` or `new_file` but there is no `close_file` nor `close_view`. One has to change the focus first and then effectively close the focused document.

Again, focus management... :(

Reply:
I hope this back and forth with this layout stuff hasn't stressed you too much.. now it's much simpler than before and it works very well in my opinion, I just couldn't imagine that this behaviour was in place.